promptsource/app.py

```diff
@@ -301,324 +145,496 @@ def get_infos(d_name):
             )
             st.stop()
 
-        template_list = dataset_templates.all_template_names
-        num_templates = len(template_list)
-        st.sidebar.write(
-            "No of prompts created for "
-            + f"`{dataset_key + (('/' + conf_option.name) if conf_option else '')}`"
-            + f": **{str(num_templates)}**"
-        )
+        #
+        # Global metrics
+        #
+        counts = template_collection.get_templates_count()
+        nb_prompted_datasets = len(counts)
+        st.write(f"## Number of *prompted datasets*: `{nb_prompted_datasets}`")
+        nb_prompts = sum(counts.values())
+        st.write(f"## Number of *prompts*: `{nb_prompts}`")
 
-        if mode == "Prompted dataset viewer":
-            if num_templates > 0:
-                template_name = st.sidebar.selectbox(
-                    "Prompt name",
-                    template_list,
-                    key="template_select",
-                    index=0,
-                    help="Select the prompt to visualize.",
-                )
+        #
+        # Metrics per dataset/subset
+        #
+        # Download dataset infos (multiprocessing download)
+        manager = Manager()
+        all_infos = manager.dict()
+        all_datasets = list(set([t[0] for t in template_collection.keys]))
+
+        pool = Pool(processes=multiprocessing.cpu_count())
+        pool.map(functools.partial(get_infos, all_infos), all_datasets)
```

Comment:
Just a side note: IIRC, even after using `multiprocessing.set_start_method("spawn", force=True)`, a newer `streamlit` itself may still cause `fork` to happen. So #411 switches to threads. Also, multithreading is faster than multiprocessing in this case.

```suggestion
        pool = ThreadPool(processes=len(all_datasets))
        pool.starmap(get_infos, zip(repeat(all_infos), all_datasets))
```

Reply:
Thanks, yeah, this was why I did something different from #411. I was still getting some errors occasionally with forking. If we want to upgrade streamlit, we could always force this pool to use spawning rather than set the global default.

Tagging in #725 .

Thanks for all your help with this!